VisualPinball.Engine/VPT/Trough/Trough.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Visual Pinball Engine
+// Copyright (C) 2020 freezy and VPE Team
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using VisualPinball.Engine.Game;
+using VisualPinball.Engine.Game.Engines;
+using VisualPinball.Engine.Math;
+
+namespace VisualPinball.Engine.VPT.Trough
+{
+	public class Trough : Item<TroughData>, ISwitchableDevice, ICoilableDevice
+	{
+		public override string ItemName { get; } = "Trough";
+		public override string ItemGroupName { get; } = null;
+
+		public const string JamSwitchId = "jam";
+		public const string EjectCoilId = "eject";
+		public const string EntryCoilId = "entry";
+
+		public IEnumerable<GamelogicEngineSwitch> AvailableSwitches => Enumerable.Repeat(0, Data.SwitchCount)
+			.Select((_, i) => new GamelogicEngineSwitch {Description = SwitchDescription(i), Id = $"{i + 1}"})
+			.Concat( new[]{ new GamelogicEngineSwitch{Description = "Jam Switch", Id = JamSwitchId} });

```

Comment:
`{` spacing doesn't seem to match. 

Const names don't really match how https://github.com/freezy/VisualPinball.Engine/blob/master/VisualPinball.Unity/VisualPinball.Unity/Game/Engine/DefaultGamelogicEngine.cs#L45 is done.

Reply:
I'll add the space :)

Don't understand your other remark about the const names. Can you elaborate?

Reply:
No big deal on this. Was just looking at `SwLeftFlipper`, `SwLeftFlipperEo`s, ie, those `const` names don't actually have the text `Id` in them.

Reply:
Oh, okay. I guess it's a bit less obvious in the `Trough` itself than in `DefaultGamelogicEngine`, where pretty much all constants are IDs.